llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h

```diff
@@ -298,7 +298,7 @@ struct CounterMappingRegion {
                        unsigned ExpandedFileID, unsigned LineStart,
                        unsigned ColumnStart, unsigned LineEnd,
                        unsigned ColumnEnd, RegionKind Kind)
-      : MCDCParams(MCDCParams), ExpandedFileID(ExpandedFileID),
+      : MCDCParams(MCDCParams), FileID(FileID), ExpandedFileID(ExpandedFileID),
```

Comment:
This is the fix for the first reported issue.

Reply:
Is this line the only bit necessary for the fix? Or are the other changes necessary?

If this is all that is necessary, then I would recommend splitting this into two patches.

Reply:
Yeah technically it is a separate fix but very minor.  FileID was not being set correctly for DecisionRegions.

Reply:
OK

I think. if we can test this, we should. But I'm fine either way.